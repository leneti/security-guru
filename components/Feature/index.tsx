export { default as Feature } from "./Feature";
export * from "./types";
