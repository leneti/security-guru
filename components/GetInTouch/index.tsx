export { default, type ContactForm } from "./GetInTouch";
