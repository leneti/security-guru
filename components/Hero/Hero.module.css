.dynamicHeight {
  min-height: var(--bg-image-height);

  @media (max-width: $mantine-breakpoint-md) {
    min-height: calc(var(--bg-image-height) - 6.25rem);
  }

  @media (max-width: $mantine-breakpoint-sm) {
    min-height: calc(var(--bg-image-height) - 6.25rem * 2);
  }
}

.hero {
  position: relative;
}

.content {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;
  z-index: 1;
  padding-bottom: calc(var(--mantine-spacing-xl) * 8);
  padding-right: clamp(
    var(--mantine-spacing-md),
    calc(100vw - var(--mantine-breakpoint-md)),
    calc(var(--mantine-breakpoint-xl) - var(--mantine-breakpoint-md))
  );
  padding-left: var(--mantine-spacing-md);

  @media (max-width: $mantine-breakpoint-xl) {
    padding-right: clamp(
      var(--mantine-spacing-md),
      calc(100vw - var(--mantine-breakpoint-md)),
      calc(
        var(--mantine-breakpoint-xl) - var(--mantine-breakpoint-md) -
          var(--hero-left-pading)
      )
    );
    padding-left: clamp(
      var(--mantine-spacing-md),
      calc(100vw - var(--mantine-breakpoint-md)),
      var(--hero-left-pading)
    );
  }

  @media (max-width: $mantine-breakpoint-md) {
    padding-right: clamp(
      var(--mantine-spacing-md),
      calc(100vw - var(--mantine-breakpoint-md)),
      calc(var(--mantine-breakpoint-xl) - var(--mantine-breakpoint-md))
    );
    padding-left: var(--mantine-spacing-md);
  }

  @media (max-width: $mantine-breakpoint-sm) {
    padding-bottom: calc(var(--mantine-spacing-xl) * 6);
  }

  @media (max-width: $mantine-breakpoint-xs) {
    padding-bottom: calc(var(--mantine-spacing-xl) * 5);
  }
}

.title {
  color: var(--mantine-color-white);
  font-size: clamp(2.125rem, 6vw, 4rem);
  font-weight: 900;
  line-height: 1.1;

  @media (max-width: $mantine-breakpoint-sm) {
    line-height: 1.2;
  }

  @media (max-width: $mantine-breakpoint-xs) {
    line-height: 1.3;
  }
}

.description {
  color: var(--mantine-color-white);
  max-width: 37.5rem;
  font-size: clamp(1rem, 2vw, 1.5rem);

  @media (max-width: $mantine-breakpoint-sm) {
    max-width: 100%;
  }
}

.CTA {
  margin-top: calc(var(--mantine-spacing-xl) * 1.5);
  color: var(--mantine-color-black);

  @media (max-width: $mantine-breakpoint-sm) {
    width: 100%;
  }
}

.image {
  object-fit: cover;
}
