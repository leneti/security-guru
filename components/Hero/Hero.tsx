import { useEffect, useState } from "react";
import {
  createStyles,
  Overlay,
  Container,
  Title,
  Button,
  Text,
  useMantineTheme,
  clsx,
} from "@mantine/core";
import Link from "next/link";
import Image from "next/image";
import { bgImgHeight } from "@site/constants";
import houses from "@site/assets/unsplash-houses.webp";

const leftPadding = "60px";
const overlayGradient =
  "linear-gradient(180deg, rgba(0, 0, 0, 0.25) 0%, rgba(0, 0, 0, .5) 40%)";

const useStyles = createStyles((theme) => ({
  dynamicHeight: {
    minHeight: bgImgHeight,

    [theme.fn.smallerThan("md")]: {
      minHeight: `calc(${bgImgHeight} - 6.25rem)`,
    },

    [theme.fn.smallerThan("sm")]: {
      minHeight: `calc(${bgImgHeight} - 6.25rem * 2)`,
    },
  },

  hero: {
    position: "relative",
  },

  content: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "flex-end",
    alignItems: "flex-start",
    zIndex: 1,
    paddingBottom: `calc(${theme.spacing.xl} * 8)`,
    paddingRight: `clamp(
      ${theme.spacing.md},
      calc(100vw - ${theme.breakpoints.md}),
      calc(${theme.breakpoints.xl} - ${theme.breakpoints.md})
    )`,
    paddingLeft: theme.spacing.md,

    [theme.fn.smallerThan("xl")]: {
      paddingRight: `clamp(
        ${theme.spacing.md},
        calc(100vw - ${theme.breakpoints.md}),
        calc(${theme.breakpoints.xl} - ${theme.breakpoints.md} - ${leftPadding})
      )`,
      paddingLeft: `clamp(
        ${theme.spacing.md},
        calc(100vw - ${theme.breakpoints.md}),
        ${leftPadding}
      )`,
    },

    [theme.fn.smallerThan("md")]: {
      paddingRight: `clamp(
        ${theme.spacing.md},
        calc(100vw - ${theme.breakpoints.md}),
        calc(${theme.breakpoints.xl} - ${theme.breakpoints.md})
      )`,
      paddingLeft: theme.spacing.md,
    },

    [theme.fn.smallerThan("sm")]: {
      paddingBottom: `calc(${theme.spacing.xl} * 6)`,
    },

    [theme.fn.smallerThan("xs")]: {
      paddingBottom: `calc(${theme.spacing.xl} * 5)`,
    },
  },

  title: {
    color: theme.white,
    fontSize: "clamp(2.125rem, 6vw, 4rem)",
    fontWeight: 900,
    lineHeight: 1.1,

    [theme.fn.smallerThan("sm")]: {
      lineHeight: 1.2,
    },

    [theme.fn.smallerThan("xs")]: {
      lineHeight: 1.3,
    },
  },

  description: {
    color: theme.white,
    maxWidth: "37.5rem",
    fontSize: "clamp(1rem, 2vw, 1.5rem)",

    [theme.fn.smallerThan("sm")]: {
      maxWidth: "100%",
    },
  },

  CTA: {
    marginTop: `calc(${theme.spacing.xl} * 1.5)`,

    [theme.fn.smallerThan("sm")]: {
      width: "100%",
    },
  },

  image: {
    objectFit: "cover",
  },
}));

export default function Hero() {
  const { classes } = useStyles();
  const theme = useMantineTheme();
  const [mounted, setMounted] = useState(false);

  useEffect(() => setMounted(true), []);

  return (
    <div className={clsx(classes.hero, classes.dynamicHeight)}>
      <Image
        src={houses}
        alt="houses"
        fill
        className={classes.image}
        priority
      />
      <Overlay
        gradient={mounted ? overlayGradient : undefined}
        opacity={mounted ? 1 : 0}
        zIndex={0}
        blur={1}
      >
        <Container
          className={clsx(classes.content, classes.dynamicHeight)}
          size={theme.breakpoints.xl}
        >
          <Title className={classes.title}>
            Namų ir verslo apsaugos sprendimai
          </Title>
          <Text className={clsx(classes.description)} mt="xl" weight={600}>
            Aukščiausio lygio paslaugos fizinės bei elektroninės apsaugos
            srityse
          </Text>

          <Button
            component={Link}
            href="/susisiekite"
            size="xl"
            radius="xl"
            className={classes.CTA}
          >
            Susisiekite su mumis
          </Button>
        </Container>
      </Overlay>
    </div>
  );
}
