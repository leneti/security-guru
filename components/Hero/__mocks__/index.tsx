export function Hero() {
  return <div>Mock Hero</div>;
}
