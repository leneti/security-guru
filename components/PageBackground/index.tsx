export { default, type PageBGProps } from "./PageBackground";
