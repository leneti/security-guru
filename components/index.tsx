export { default as ContactInfo } from "./ContactInfo";
export { default as Footer } from "./Footer";
export { default as GetInTouch, type ContactForm } from "./GetInTouch";
export { default as Header } from "./Header";
export { default as Hero } from "./Hero";
export { default as Logo } from "./Logo";
export { default as PageTitle } from "./PageTitle";
export { default as PageBackground, type PageBGProps } from "./PageBackground";
export { default as SlideDownTitle } from "./SlideDownTitle";
