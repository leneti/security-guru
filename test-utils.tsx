import type { ReactElement, ReactNode } from "react";
import { MantineProvider } from "@mantine/core";
import { render, type RenderOptions } from "@testing-library/react";
import { resolver, theme } from "@site/constants";

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return (
    <MantineProvider theme={theme} cssVariablesResolver={resolver}>
      {children}
    </MantineProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
