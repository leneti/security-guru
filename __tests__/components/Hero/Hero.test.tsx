import { render } from "@site/test-utils";
import { Hero } from "@site/components/Hero";

describe("Hero", () => {
  it("renders", () => {
    const { container } = render(<Hero />);
    expect(container).toMatchSnapshot();
  });
});
