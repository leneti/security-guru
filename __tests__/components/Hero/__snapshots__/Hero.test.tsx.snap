// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders 1`] = `
<div>
  <div
    class="mantine-1iugybl mantine-1y9xq1b"
  >
    <img
      alt="houses"
      class="mantine-18szc2b"
      data-nimg="fill"
      decoding="async"
      sizes="100vw"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <div
      class="mantine-Overlay-root mantine-1pa6mid"
    >
      <div
        class="mantine-Container-root mantine-kypcyk"
      >
        <h1
          class="mantine-Text-root mantine-Title-root mantine-1pa9w1p"
        >
          Namų ir verslo apsaugos sprendimai
        </h1>
        <div
          class="mantine-Text-root mantine-1bm3ovv"
        >
          Aukščiausio lygio paslaugos fizinės bei elektroninės apsaugos srityse
        </div>
        <a
          class="mantine-UnstyledButton-root mantine-Button-root mantine-hb0fps"
          data-button="true"
          href="/susisiekite"
          type="button"
        >
          <div
            class="mantine-1wpc1xj mantine-Button-inner"
          >
            <span
              class="mantine-1ryt1ht mantine-Button-label"
            >
              Susisiekite su mumis
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
